AirSim/AirLib/include/common/AirSimSettings.hpp

```diff
@@ -1196,8 +1196,13 @@ struct AirSimSettings {
         case SensorBase::SensorType::Lidar:
             sensor_setting = std::unique_ptr<SensorSetting>(new LidarSetting());
             break;
+        case SensorBase::SensorType::GSS:
+            // at this moment the codebase requires every sensor to have settings.
+            // However, gss does not have settings... so we just place ImuSettings in here for now.
+            sensor_setting = std::unique_ptr<SensorSetting>(new ImuSetting());
```

Comment:
how does this work? Can you give an example of a settings.json that has an IMU and a GSS?

Reply:
The `new ImuSetting()` doesn't do anything with the imu settings in the json, it just creates an empty settings object.

The problem is that `sensor_setting ` needs to have a valid object and I don't want to create the `GSSSettings` class because gss doesn't have settings. The good solution would be to change the abstraction structure to allow sensors without settings. But I don't want to do that because lotsofwork and notthatusefullreally.